dbt_copilot_helper/templates/addons/env/aurora-postgres.yml

```diff
@@ -151,3 +152,309 @@ Resources:
+              copilot_environment = event['ResourceProperties']['CopilotEnvironment']
+
+              application_user_secret = None
+
+              try:
+                  application_user_secret = secrets_manager.create_secret(
+                      Name=application_user_secret_name,
+                      Description=application_user_secret_description,
+                      SecretString=json.dumps(application_user_secret_string),
+                      Tags=[
+                          {'Key': 'custom:cloudformation:stack-name', 'Value': event["StackId"].split('/')[1]},
+                          {'Key': 'custom:cloudformation:logical-id', 'Value': event["LogicalResourceId"]},
+                          {'Key': 'custom:cloudformation:stack-d', 'Value': event["StackId"]},
+                          {'Key': 'copilot-application', 'Value': copilot_application},
+                          {'Key': 'copilot-environment', 'Value': copilot_environment},
+                      ],
+                  )
+              except ClientError as error:
+                  if error.response["Error"]["Code"] == "ResourceExistsException":
+                      application_user_secret = secrets_manager.update_secret(
+                          SecretId=application_user_secret_name,
+                          SecretString=json.dumps(application_user_secret_string),
+                      )
+
+              return application_user_secret
+
+
+          def handler(event, context):
+              print("REQUEST RECEIVED:\n" + json.dumps(event))
+
+              db_master_user_secret = event['ResourceProperties']['MasterUserSecret']
+              application_user_secret_name = event['ResourceProperties']['SecretName']
+              application_user_secret_description = event['ResourceProperties']['SecretDescription']
+              copilot_application = event['ResourceProperties']['CopilotApplication']
+              copilot_environment = event['ResourceProperties']['CopilotEnvironment']
+
+              secrets_manager = boto3.client("secretsmanager")
+
+              master_user = json.loads(secrets_manager.get_secret_value(SecretId=db_master_user_secret)["SecretString"])
+
+              application_user_password = secrets_manager.get_random_password(
+                  PasswordLength=16,
+                  ExcludeCharacters='[]{}()"@/\;=?&`><:|#',
+                  ExcludePunctuation=True,
+                  IncludeSpace=False,
+              )["RandomPassword"]
+
+              application_user_secret_string = {
+                  "username": "application_user",
+                  "password": application_user_password,
+                  "engine": master_user["engine"],
+                  "port": master_user["port"],
+                  "dbname": master_user["dbname"],
+                  "host": master_user["host"],
+                  "dbClusterIdentifier": master_user["dbClusterIdentifier"]
```

Comment:
This is the only line which differs in the code of the two lambdas.

For a separate pull request, I wonder if we can publish method in `dbt-copilot-python` that takes a parameter to specify the string used above and reduce the duplication?

Reply:
We would need to publish the code to an S3 bucket, but we can also retrieve this info through the Custom Resource and use a `try/except` block. Not sure how useful `dbt-copilot-python` would be, unless we install the package in the `-deploy` repositories as well...